test/lib/ufe/testConnections.py

```diff
@@ -951,35 +951,67 @@ def testCreateNodeGraphAttributes(self):
             testAttrs.addObserver(testItem, testObserver)
             testObserver.assertNotificationCount(self)
 
-            testAttrs.addAttribute("inputs:foo", ufe.Attribute.kFloat)
+            testAttrs.addAttribute("inputs:foo", ufe.Attribute.kFloat4)
             testObserver.assertNotificationCount(self, numAdded = 1)
-            testAttrs.addAttribute("outputs:bar", ufe.Attribute.kFloat4)
+            testAttrs.addAttribute("outputs:bar", ufe.Attribute.kFloat)
             testObserver.assertNotificationCount(self, numAdded = 2)
-            
+            testAttrs.addAttribute("inputs:enumString", ufe.Attribute.kString)
+            testObserver.assertNotificationCount(self, numAdded = 3)
+            enumAttr = testAttrs.attribute("inputs:enumString")
+            # Not yet an enum:
+            self.assertEqual(enumAttr.type, ufe.Attribute.kString)
+            # But with proper metadata:
+            enumAttr.setMetadata("enum", "foo, bar, baz")
+            enumAttr = testAttrs.attribute("inputs:enumString")
+            # It is now an enum:
+            self.assertEqual(enumAttr.type, ufe.Attribute.kEnumString)
+            # Test the other enum API:
+            if hasattr(testAttrs, "getEnums"):
+                enums = testAttrs.getEnums("inputs:enumString")
+                self.assertEqual(len(enums), 3)
+                self.assertEqual(len(enums[0]), 2)
+                self.assertEqual(enums[0][0], "foo")
```

Comment:
Should you be testing bar and baz too?

Reply:
I am confident the other two labels are correct. There is also a complete check at line 986.